frontends/ol-components/src/components/LearningResourceExpanded/LearningResourceExpandedV2.tsx

```diff
@@ -413,55 +502,132 @@ const CallToActionSection = ({
-        {platformImage ? (
-          <Platform>
-            <OnPlatform>on</OnPlatform>
-            <StyledPlatformLogo platformCode={platformCode} height={26} />
-          </Platform>
-        ) : null}
-        <ListButtonContainer>
+      <ActionsContainer>
+        <StyledLink
+          target="_blank"
+          size="medium"
+          data-ph-action="click-cta"
+          data-ph-offered-by={offeredBy?.code}
+          data-ph-resource-type={resource.resource_type}
+          data-ph-resource-id={resource.id}
+          endIcon={<RiExternalLinkLine />}
+          href={getCallToActionUrl(resource) || ""}
+        >
+          {cta}
+        </StyledLink>
+        <PlatformContainer>
+          {platformImage ? (
+            <Platform>
+              <OnPlatform>on</OnPlatform>
+              <StyledPlatformLogo platformCode={platformCode} height={26} />
+            </Platform>
+          ) : null}
+        </PlatformContainer>
+        <ButtonContainer>
           {user?.is_learning_path_editor && (
-            <CardActionButton
-              filled={inLearningPath}
-              aria-label="Add to Learning Path"
+            <CallToActionButton
+              selected={inLearningPath ? 1 : 0}
+              startIcon={<RiMenuAddLine />}
+              aria-label={addToLearningPathLabel}
               onClick={(event) =>
                 onAddToLearningPathClick
                   ? onAddToLearningPathClick(event, resource.id)
                   : null
               }
             >
-              <RiMenuAddLine aria-hidden />
-            </CardActionButton>
+              {addToLearningPathLabel}
+            </CallToActionButton>
           )}
-          <CardActionButton
-            filled={inUserList}
-            aria-label={`Bookmark ${getReadableResourceType(resource.resource_type)}`}
+          <CallToActionButton
+            selected={inUserList ? 1 : 0}
+            startIcon={inUserList ? <RiBookmarkFill /> : <RiBookmarkLine />}
+            aria-label={bookmarkLabel}
             onClick={
               onAddToUserListClick
                 ? (event) => onAddToUserListClick?.(event, resource.id)
                 : undefined
             }
           >
-            <RiBookmarkLine aria-hidden />
-          </CardActionButton>
-        </ListButtonContainer>
-      </PlatformContainer>
+            {bookmarkLabel}
+          </CallToActionButton>
+          <CallToActionButton
+            selected={shareExpanded ? 1 : 0}
+            startIcon={<RiShareLine />}
+            aria-label={shareLabel}
+            onClick={() => setShareExpanded(!shareExpanded)}
+          >
+            {shareLabel}
+          </CallToActionButton>
+        </ButtonContainer>
+        {shareExpanded && shareUrl && (
+          <ShareContainer data-testid="drawer-share">
+            <ShareLabel>Share a link to this Resource</ShareLabel>
+            <Input
+              fullWidth
+              ref={inputRef}
```

Comment:
```suggestion
              inputRef={inputRef}
```
would simplify the stuff below.

Actually, though: You don't need a ref at all. The click handler could be
```ts
(event) => {
  const insputEl = event.currentTarget.querySelector("input")
  if (!inputEl) return
  input.select()
}